nemo/collections/nlp/models/dialogue/intent_bio_type_classification_model.py

```diff
@@ -0,0 +1,900 @@
+
+        for one_mention_mask, one_hidden_states in zip(mention_mask, hidden_states):
+            mention_states = []
+            for idx, one_mask_value in enumerate(one_mention_mask):
+                if one_mask_value == 1:
+                    mention_states.append([])
+                if one_mask_value in [1, 2]:
+                    mention_states[-1].append(one_hidden_states[idx])
+            
+            mention_states = [torch.mean(torch.stack(mention_state),0) for mention_state in mention_states]
+            mention_states += [torch.zeros((hidden_states_dim)).to(hidden_states.device) for i in range(max_token_len-len(mention_states))]
+            mention_hidden_states.append(torch.stack(mention_states))
+        
+        return torch.stack(mention_hidden_states)
+
+    @typecheck()
+    def forward(self, input_ids, attention_mask, token_type_ids, mention_mask, mention_loss_mask):
+        """
+        No special modification required for Lightning, define it as you normally would
+        in the `nn.Module` in vanilla PyTorch.
+        """
+        if self._cfg.tokenizer.get('library', '') == 'megatron':
+            hidden_states, _ = self.bert_model(input_ids, attention_mask, tokentype_ids=token_type_ids, lm_labels=None)
+        else:
+            hidden_states = self.bert_model(
+                input_ids=input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+            )
+
+        from os.path import exists
+        global UNIQUE_EPOCH_COUNTER
+
+        # print("START DUMP hidden_states")
+        # path_to_file="/home/lilee/pickle/hidden_states"+str(UNIQUE_EPOCH_COUNTER)+".pickle"
+        # if exists(path_to_file)==False:
+        #     with open(path_to_file, 'wb') as f:
+        #         # Pickle the 'data' dictionary using the highest protocol available.
+        #         pickle.dump(hidden_states, f, pickle.HIGHEST_PROTOCOL)        
+        # print("END DUMP hidden_states")
+
+
+        # intent_logits, slot_logits = self.classifier(hidden_states=hidden_states)
+        intent_logits, bio_slot_logits = self.classifier(hidden_states=hidden_states)
+
+        # print("START DUMP bio_slot_logits")
+        # path_to_file="/home/lilee/pickle/bio_slot_logits"+str(UNIQUE_EPOCH_COUNTER)+".pickle"
+        # if exists(path_to_file)==False:
+        #     with open(path_to_file, 'wb') as f:
+        #         # Pickle the 'data' dictionary using the highest protocol available.
+        #         pickle.dump(bio_slot_logits, f, pickle.HIGHEST_PROTOCOL)
+        # print("END DUMP bio_slot_logits")
+
+        # current:bug; hidden_states#(batch*128*768) zzz
+        
+        # mention_hidden_states_pad: b*128*hiddenstate_dim
+        mention_hidden_states_pad = IntentBIOTypeClassificationModel.get_entity_embedding_from_hidden_states(mention_mask, hidden_states)
+        
+        """
+        RUN TIME error cause here (may because too many for loop, may have better way to implement):
```

Comment:
I agree - I think using tensor operations might be better and makes it differentiable, which can allow it to be trained with backpropogation. Do you know which line exactly is the process killed? I'm also struggling to understand what exactly you're trying to do here. Maybe you can walk me through it again?